### PR TITLE
1148: course-enrollment-upgrading-is-not-ever-synchronized-with-edx-if-the-original-update-request-fails

### PR DESCRIPTION
**Pull request by collinpreston**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
 
#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/1148

#### What's this PR do?
Sets the course_run_enrollment's `edx_enrolled` value equal to False when the edx API call, to modify an existing course_enrollment record's `mode` from `audit` to `verified`, fails.  This ensures that the edx API call is attempted again when the `retry-failed-edx-enrollments` task is performed periodically.  Ultimately this ensure that we synchronize the enrollment mode between `mitxonline` and edx even if the edx API calls fail.

#### How should this be manually tested?
Tester should have `devstack` and `mitxonline` running locally.

- Before starting `mitxonline`, I modified this section [here](https://github.com/mitodl/mitxonline/blob/7f4246df904f3557e28f3c662f2780bfd810b100/main/settings.py#L797-L801) so that the task to sync course modes runs more often.   
- Create a course, course_run, associated product, associated course CMS page all within `mitxonline`.  To make things easier, I used the default course that is created in `devstack` ([E2E Test Course](http://localhost:18000/courses/course-v1:edX+E2E-101+course).
- Enroll your user through `mitxonline` into the free version of the course.
- Ensure that both `mitxonline` and `devstack` show a course mode value for the user's enrollment with a value of `audit`.
- Perform a `docker-compose stop` on your `devstack`.
- Upgrade your user's enrollment through `mitxonline` by paying for the course.
- Perform a `make dev.up.large-and-slow on your `devstack` to start it again.
- Wait until the `retry_failed_edx_enrollments` task is run again.  You should be able to see this mention in the `celery` logs for `mitxonline`.
- Once the `retry_failed_edx_enrollments` task is run, verify the course mode for the enrollment on `devstack` has changed from `audit` to `verified`.

### CODE REVIEW

File: courses/api.py [hunk 1]
```diff
@@ -275,6 +275,7 @@ def send_enrollment_emails():
                 # Case (Upgrade): When user was enrolled in free mode and now enrolls in paid mode (e.g. Verified)
                 # So, User has an active enrollment and the only changing thing is going to be enrollment mode
                 if enrollment.active and enrollment_mode_changed:
+                    enrollment.edx_enrolled = edx_request_success
```
Comment (rachellougee) [inline courses/api.py:278]: This line could be moved to line 275  as if and elif blocks now have the same exact line

File: courses/api_test.py [hunk 2]
```diff
@@ -30,6 +30,11 @@
     ENROLL_CHANGE_STATUS_DEFERRED,
     ENROLL_CHANGE_STATUS_REFUNDED,
 )
+from openedx.constants import (
+    EDX_DEFAULT_ENROLLMENT_MODE,
+    EDX_ENROLLMENT_AUDIT_MODE,
+    EDX_ENROLLMENT_VERIFIED_MODE,
```
Comment (rachellougee) [inline courses/api_test.py:36]: There is already import on line 51

**Review by rachellougee — COMMENTED** [review]
Just small comments. I will do a functional test

File: courses/api.py (hunk 1 above)
Comment (collinpreston) [reply courses/api.py]: Nice!  Good catch

**Review by rachellougee — APPROVED** [review]
LGTM

**Pull request merged**